app/bundles/CoreBundle/Templating/Twig/Extension/MenuExtension.php

```diff
@@ -37,4 +41,49 @@ public function menuRender($menu, array $options = [], ?string $renderer = null)
     {
         return $this->menuHelper->render($menu, $options, $renderer);
     }
+
+    /**
+     * Parses attributes for the menu view.
+     *
+     * @param array<string> $attributes
+     * @param array<string> $overrides
+     */
+    public function parseMenuAttributes($attributes, $overrides = []): string
+    {
+        return $this->menuHelper->parseAttributes($attributes, $overrides);
+    }
+
+    /**
+     * Concats the appropriate classes for menu links.
+     *
+     * @param ItemInterface|null    $item
+     * @param MatcherInterface|null $matcher
+     * @param array<string,string>  $options
+     * @param string                $extra
+     *
+     * @return array<mixed>
+     */
+    public function buildMenuClasses($item, $matcher, $options, $extra)
```

Comment:
```suggestion
    /**
     * Concats the appropriate classes for menu links.
     *
     * @param array<string,string>  $options
     *
     * @return array<mixed>
     */
    public function buildMenuClasses(ItemInterface $item, MatcherInterface $matcher, array $options, ?string $extraClass = null) : array
```
I'm fuzzy about the nullable params. But from the implementation it looks like the first 2 must be required otherwise it will fail. I suggest to also rename the last param to better describe what `extra` means. I'd even considered if there must be a forth param. Why it's not part of the `$options`?

Reply:
I made some changes to the function to handle things like null, that should resolve the issue of the nullable params.
I've also changed the naming on $extra and inlined the param types as suggested